### PR TITLE
minor refactor of saml_settings

### PR DESCRIPTION
**Pull request by saneshark**
- this change is to support further refactor of sessions_controller
- moving persist_session_and_user into a class for better handling and testing in future PR.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -125,15 +125,7 @@ def render_unauthorized
   end
 
   def saml_settings(options = {})
```
Comment (kreek) [inline app/controllers/application_controller.rb]: Other than providing a default option for options, the method just seems to delegate to SAML::SettingsService could the wrapper go away?

Comment (saneshark) [reply app/controllers/application_controller.rb]: it totally could, I was just being lazy about rewriting specs that currently depend on this controller method, but by moving the logic in models, parts of session_controller could immedietely be refactored to use SAML::SettingsService.saml_settings(with_whatever_options).

Comment (saneshark) [reply app/controllers/application_controller.rb]: I can make that change to controllers and controller specs though if you prefer, but figured they'd naturally go away with future refactors.

Comment (kreek) [reply app/controllers/application_controller.rb]: 👍  either way works, LGTM 🚀

**Pull request merged**